tests/unit-core/test-symbol.c

```diff
@@ -0,0 +1,196 @@
+  jerry_value_t value_3 = jerry_create_string (STRING_BAR);
+
+  prop_desc.is_value_defined = true;
+  prop_desc.value = jerry_acquire_value (value_3);
+  prop_desc.is_writable_defined = true;
+  prop_desc.is_writable = false;
+  prop_desc.is_enumerable_defined = true;
+  prop_desc.is_enumerable = false;
+  prop_desc.is_configurable_defined = true;
+  prop_desc.is_configurable = false;
+  TEST_ASSERT (jerry_get_boolean_value (jerry_define_own_property (object, symbol_2, &prop_desc)));
+  jerry_free_property_descriptor_fields (&prop_desc);
+
+  /* Check the modified fields */
+  TEST_ASSERT (jerry_get_own_property_descriptor (object, symbol_2, &prop_desc));
+  TEST_ASSERT (prop_desc.is_value_defined == true);
+  TEST_ASSERT (value_3 == prop_desc.value);
+  TEST_ASSERT (jerry_value_is_string (prop_desc.value));
+  TEST_ASSERT (prop_desc.is_writable_defined == true);
+  TEST_ASSERT (prop_desc.is_writable == false);
+  TEST_ASSERT (prop_desc.is_enumerable_defined == true);
+  TEST_ASSERT (prop_desc.is_enumerable == false);
+  TEST_ASSERT (prop_desc.is_configurable_defined == true);
+  TEST_ASSERT (prop_desc.is_configurable == false);
+  TEST_ASSERT (prop_desc.is_get_defined == false);
+  TEST_ASSERT (jerry_value_is_undefined (prop_desc.getter));
+  TEST_ASSERT (prop_desc.is_set_defined == false);
+  TEST_ASSERT (jerry_value_is_undefined (prop_desc.setter));
+  jerry_free_property_descriptor_fields (&prop_desc);
+
+  jerry_release_value (value_3);
+  jerry_release_value (value_2);
+  jerry_release_value (symbol_2);
+  jerry_release_value (object);
+
+  /* Test creating symbol with a symbol description */
+  jerry_value_t empty_symbol_desc = jerry_create_string ((const jerry_char_t *) "");
+
+  jerry_value_t empty_symbol = jerry_create_symbol (empty_symbol_desc);
+  TEST_ASSERT (!jerry_value_is_error (empty_symbol));
+  TEST_ASSERT (jerry_value_is_symbol (empty_symbol));
+
+  jerry_release_value (empty_symbol_desc);
+
+  jerry_value_t symbol_symbol = jerry_create_symbol (empty_symbol);
+  TEST_ASSERT (!jerry_value_is_symbol (symbol_symbol));
+  TEST_ASSERT (jerry_value_is_error (symbol_symbol));
+
+  jerry_value_t error_obj = jerry_get_value_from_error (symbol_symbol, true);
+
+  TEST_ASSERT (jerry_get_error_type (error_obj) == JERRY_ERROR_TYPE);
+
+  jerry_release_value (error_obj);
+  jerry_release_value (empty_symbol);
+
+  /* Test symbol to string operation with symbol argument */
+  jerry_value_t bar_symbol_desc = jerry_create_string (STRING_BAR);
+
+  jerry_value_t bar_symbol = jerry_create_symbol (bar_symbol_desc);
+  TEST_ASSERT (!jerry_value_is_error (bar_symbol));
+  TEST_ASSERT (jerry_value_is_symbol (bar_symbol));
+
+  jerry_release_value (bar_symbol_desc);
+
+  jerry_value_t bar_symbol_string = jerry_get_symbol_descriptive_string (bar_symbol);
+  TEST_ASSERT (jerry_value_is_string (bar_symbol_string));
+
+  jerry_size_t bar_symbol_string_size = jerry_get_string_size (bar_symbol_string);
```

Comment:
The `bar_symbol_string_size` should also be checked if it is of the expected size.

Reply:
Good idea!